connection/connection.ts

```diff
@@ -255,56 +255,60 @@ export class Connection {
       hostname,
       port,
       tls: {
+        enabled: isTLSEnabled,
         enforce: enforceTLS,
       },
     } = this.#connection_params;
 
     // A BufWriter needs to be available in order to check if the server accepts TLS connections
     await this.#createNonTlsConnection({ hostname, port });
 
-    const accepts_tls = await this.#serverAcceptsTLS()
-      .catch((e) => {
-        // Make sure to close the connection if the TLS validation throws
-        this.#conn.close();
-        throw e;
-      });
+    // If TLS is disabled, we don't even try to connect.
+    if (isTLSEnabled) {
```

Comment:
Per spec, all PostgreSQL servers with TLS encryption support (all supported versions) must start a connection asking whether of not the server is willing to create a TLS connection, even if the user requested TLS to be disabled. Simply move the "is TLS enabled" condition after the serverAcceptsTLS check to resolve this